java/ql/src/experimental/Security/CWE/CWE-749/UnsafeAndroidAccess.ql

```diff
@@ -0,0 +1,138 @@
+/**
+ * @name Unsafe resource fetching in Android webview
+ * @description JavaScript rendered inside WebViews can access any protected application file and web resource from any origin
+ * @kind path-problem
+ * @tags security
+ *       external/cwe/cwe-749
+ *       external/cwe/cwe-079
+ */
+
+import java
+import semmle.code.java.frameworks.android.Intent
+import semmle.code.java.frameworks.android.WebView
+import semmle.code.java.dataflow.FlowSources
+
+/**
+ * Methods allowing any-local-file and cross-origin access in the WebSettings class
+ */
+class CrossOriginAccessMethod extends Method {
+  CrossOriginAccessMethod() {
+    this.getDeclaringType() instanceof TypeWebSettings and
+    (
+      this.hasName("setAllowUniversalAccessFromFileURLs") or
+      this.hasName("setAllowFileAccessFromFileURLs")
+    )
+  }
+}
+
+/**
+ * `setJavaScriptEnabled` method for the webview
+ */
+class AllowJavaScriptMethod extends Method {
+  AllowJavaScriptMethod() {
+    this.getDeclaringType() instanceof TypeWebSettings and
+    this.hasName("setJavaScriptEnabled")
+  }
+}
+
+/**
+ * Holds if a call to `v.setJavaScriptEnabled(true)` exists
+ */
+predicate isJSEnabled(Variable v) {
+  exists(MethodAccess jsa |
+    v.getAnAccess() = jsa.getQualifier() and
+    jsa.getMethod() instanceof AllowJavaScriptMethod and
+    jsa.getArgument(0).(BooleanLiteral).getBooleanValue() = true
+  )
+}
+
+/**
+ * Fetch URL method call on the `android.webkit.WebView` object
+ */
+class FetchResourceMethodAccess extends MethodAccess {
+  FetchResourceMethodAccess() {
+    this.getMethod().getDeclaringType() instanceof TypeWebView and
+    this.getMethod().hasName(["loadUrl", "postUrl"])
+  }
+}
+
+/**
+ * Method access to external inputs of `android.content.Intent` object
+ */
+class IntentGetExtraMethodAccess extends MethodAccess {
+  IntentGetExtraMethodAccess() {
+    this.getMethod().getName().regexpMatch("get\\w+Extra") and
+    this.getMethod().getDeclaringType() instanceof TypeIntent
+    or
+    this.getMethod().getName().regexpMatch("get\\w+") and
+    this.getQualifier().(MethodAccess).getMethod().hasName("getExtras") and
+    this.getQualifier().(MethodAccess).getMethod().getDeclaringType() instanceof TypeIntent
+  }
+}
+
+/**
+ * Source of fetching urls
+ */
+class UntrustedResourceSource extends RemoteFlowSource {
+  UntrustedResourceSource() {
+    exists(IntentGetExtraMethodAccess ma |
+      this.asExpr().(VarAccess).getVariable().getAnAssignedValue() = ma
+    )
+  }
+
+  override string getSourceType() { result = "UntrustedIntentExtraSource" }
```

Comment:
With the above change this override is no longer needed (nor indeed valid).
```suggestion
```

Reply:
Removed the override and updated the comments